proto/gnmi_ext/gnmi_ext.proto

```diff
@@ -89,3 +92,42 @@ message TimeRange {
   int64 start = 1; // Nanoseconds since the epoch
   int64 end = 2;   // Nanoseconds since the epoch
 }
+
+// Commit confirmed extension allows automated revert of the configuration after
+// certain duration if an explicit confirmation is not issued. It allows explicit
+// cancellation of the commit during the rollback window. There cannot be more
+// than one commit active at a given time.
+// The document about gNMI commit confirmed can be found at
+// https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-commit-confirmed.md
+message Commit {
+  // ID is provided by the client during the commit request. During confirm and cancel
+  // actions the provided ID should match the ID provided during commit.
+  // If ID is not passed in any actions server shall return error.
+  // Required.
+  string id = 1;
+  oneof action {
+    // commit action creates a new commit. If a commit is on-going, server returns error.
+    CommitRequest commit = 2;
+    // confirm action will confirm an on-going commit, the ID provided during confirm
+    // should match the on-going commit ID.
+    CommitConfirm confirm = 3;
+    // cancel action will cancel an on-going commit, the ID provided during cancel
+    // should match the on-going commit ID.
+    CommitCancel cancel = 4;
+  }
+}
```

Comment:
@kidiyoor, to pick up on our discussion in OC community call: We at Nokia can support this mode of operation where a client chooses the ID for he commit confirm operation.

I wonder, though, if we should consider adding `CommitResponse` message that clients can provide in the SetResponse extension field to allow servers to respond with custom data that clients MAY be interested in.
This data may be `bytes` and can contain something like internal commit ID that we discussed.

Again, this `CommitResponse` message is optional, servers might choose not to set it up at all.

WDYT?

Reply:
I would be careful with introducing a field which don't have a standard behavior across vendors in the spec. Since the goal of gnmi is to standardize the API across vendors, with such an optional response there is possibility of client/server implementation which is not interoperable.

If we can define a common use-case, with dedicated fields(maybe not bytes), then I think we can definitely consider.

Reply:
Can you talk more about the use-case here (is it same as the subscription use-case that was mentioned during community call)?

If the data expected form the server is going to be server generated ID - then is it only relevant to "confirmed commit" use-case ? should the response data be part of regular Set commit as well ? can you talk about why its only specific to "confirmed commit" ?

Reply:
@kidiyoor I was just thinking that returning back internal ID might be useful down the road.
But I am ok dropping this and us (nokia) using a custom extension instead.

Reply:
@hellt if you can describe the `CommitResponse` message in proto, then I think this could be modeled. I think this would make sense to in a follow up PR, unless you think it would conflict with this PR as is.

Reply:
@dplore yes, we can follow up in the next PR